File: Web.Hubs/Web.Hubs.Core/Entities/Message.cs
namespace Web.Hubs.Core.Entities;

public sealed class Message
{
    public Guid Id { get; init; }

    public required string Content { get; init; }

    public required long UserId { get; init; }

    public DateTime Date { get; init; } = DateTime.UtcNow;

    public bool Edit { get; set; }

    public required Guid ChatId { get; init; }

    public Chat? Chat { get; init; }
}

File: Web.UI/src/types/message.ts
export type Message = {
    id: string,
    content: string,
    userId: number,
    chatId: string,
    date: string,
    edit: boolean
}

export type LastMessage = {
    date: string,
    content: string
}

export type CreateMessage = {
    chatId: string,
    content: string
}